server.go

```diff
@@ -409,6 +409,16 @@ func (s *ServerImpl) SetMainByPort(port int) error {
 	return s.reloadHaproxy(port)
 }
 
+func (s *ServerImpl) SetMainById(id string) error {
+	for port, deployId := range s.config.Ports {
+		if deployId == id {
+			return s.reloadHaproxy(port)
```

Comment:
Perhaps this should call SetMainByPort(port) instead

Then if we choose to implement a check that something's actually running on that port...